src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -136,66 +109,106 @@ private enum BatchWriterState
+                Debug.Assert(this.currentOperationRequestMessage == null || !this.outputContext.WritingResponse, "Request message can only be filled when writing request.");
                 Debug.Assert(this.currentOperationRequestMessage == null || this.currentOperationResponseMessage == null, "Only request or reponse message can be set, not both.");
                 return this.currentOperationRequestMessage;
             }
 
             set
             {
-                Debug.Assert(value == null || !this.rawOutputContext.WritingResponse, "Can only set the request message if we're writing a request.");
+                Debug.Assert(value == null || !this.outputContext.WritingResponse, "Can only set the request message if we're writing a request.");
                 Debug.Assert(this.currentOperationRequestMessage == null || this.currentOperationResponseMessage == null, "Only request or reponse message can be set, not both.");
                 this.currentOperationRequestMessage = value;
             }
         }
 
         /// <summary>The response message for the operation that is currently written if it's a response;
         /// or null if no operation is written right now or it's a request operation.</summary>
-        private ODataBatchOperationResponseMessage CurrentOperationResponseMessage
+        protected ODataBatchOperationResponseMessage CurrentOperationResponseMessage
         {
             get
             {
-                Debug.Assert(this.currentOperationResponseMessage == null || this.rawOutputContext.WritingResponse, "Response message can only be filled when writing response.");
+                Debug.Assert(this.currentOperationResponseMessage == null || this.outputContext.WritingResponse, "Response message can only be filled when writing response.");
                 Debug.Assert(this.currentOperationRequestMessage == null || this.currentOperationResponseMessage == null, "Only request or reponse message can be set, not both.");
                 return this.currentOperationResponseMessage;
             }
 
             set
             {
-                Debug.Assert(value == null || this.rawOutputContext.WritingResponse, "Can only set the response message if we're writing a response.");
+                Debug.Assert(value == null || this.outputContext.WritingResponse, "Can only set the response message if we're writing a response.");
                 Debug.Assert(this.currentOperationRequestMessage == null || this.currentOperationResponseMessage == null, "Only request or reponse message can be set, not both.");
                 this.currentOperationResponseMessage = value;
             }
         }
 
-        /// <summary>The message for the operation that is currently written; or null if no operation is written right now.</summary>
-        private ODataBatchOperationMessage CurrentOperationMessage

```

Comment:
>private ODataBatchOperationMessage CurrentOperationMessage [](start = 8, length = 58)

Changing the order of code blocks makes it harder for code review. For example this methods with one change from private to internal moved down. But it shows here as deleted code and down as added code. If you could keep the current alignment it would be easier for CR.

Reply:
StyleCop requires that interval variables should be placed after private ones.